src/main/java/fr/ens/biologie/genomique/eoulsan/modules/diffana/DESeq2Checker.java

```diff
@@ -0,0 +1,81 @@
+package fr.ens.biologie.genomique.eoulsan.modules.diffana;
+
+import fr.ens.biologie.genomique.eoulsan.EoulsanException;
+import fr.ens.biologie.genomique.eoulsan.design.Experiment;
+import fr.ens.biologie.genomique.eoulsan.design.ExperimentMetadata;
+
+import static fr.ens.biologie.genomique.eoulsan.design.DesignUtils.getAllSamplesMetadataKeys;
+import static fr.ens.biologie.genomique.eoulsan.design.DesignUtils.getExperimentSampleAllMetadataKeys;
+import static fr.ens.biologie.genomique.eoulsan.design.SampleMetadata.CONDITION_KEY;
+import static fr.ens.biologie.genomique.eoulsan.design.SampleMetadata.REP_TECH_GROUP_KEY;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class is made to check the design.txt file before running DESeq2.
+ * @author Charlotte Berthelier
+ * @since 2.4
+ */
+
+public class DESeq2Checker {
+
+    /**
+     * Check experiment design.
+     * @param experiment experiment to check
+     * @throws EoulsanException if the experiment design is not correct
+     */
+    static boolean checkExperimentDesign(Experiment experiment) throws EoulsanException {
+        return checkExperimentDesign (experiment, true);
+    }
+
+    /**
+     * Check experiment design.
+     * @param experiment experiment to check
+     * @param throwsException if true throw an exception
+     * @throws EoulsanException if the experiment design is not correct
+     */
+    static boolean checkExperimentDesign(Experiment experiment, boolean throwsException) throws EoulsanException {
+
+        requireNonNull(experiment, "Experiment argument cannot be null");
+
+        final ExperimentMetadata emd = experiment.getMetadata();
+
+        if (emd.containsComparisons()) {
+
+            // Check if the comparison value is correct
+            for (String c : emd.getComparisons().split(";")) {
+                String[] splitC = c.split(":");
```

Comment:
splitC[0] (after trimming) cannot be empty
You can also use Guava's spliter class to easily split your strings:
Splitter.on(':').omitEmptyStrings().trimResults().splitToList(c) -> Geta list of trimmed elements without empty elements